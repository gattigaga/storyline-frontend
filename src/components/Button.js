import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Content = styled.button`
  padding: 12px 16px;
  background: ${({ isOutlined }) => (isOutlined ? "white" : "#e74c3c")};
  color: ${({ isOutlined }) => (isOutlined ? "#e74c3c" : "white")};
  border: 1px solid ${({ isOutlined }) => (isOutlined ? "#e74c3c" : "white")};
  font-family: Roboto;
  font-size: 14px;
  cursor: pointer;
  border-radius: 4px;
  outline: none;
`;

const Button = ({ className, caption, isOutlined, onClick }) => (
  <Content className={className} isOutlined={isOutlined} onClick={onClick}>
    {caption}
  </Content>
);

Button.propTypes = {
  caption: PropTypes.string,
  isOutlined: PropTypes.bool,
  className: PropTypes.string,
  onClick: PropTypes.func
};

Button.defaultProps = {
  caption: "Caption"
};

export default Button;
