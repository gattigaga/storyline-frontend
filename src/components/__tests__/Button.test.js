import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Button from "../Button";

describe("Button", () => {
  const setup = propOverrides => {
    const props = {
      caption: "Login",
      onClick: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Button {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders with outline", () => {
    const { wrapper } = setup({ isOutlined: true });

    expect(wrapper.props().isOutlined).toEqual(true);
  });

  it("should call 'onClick' callback when pressed", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("click");

    expect(props.onClick).toBeCalled();
  });
});
