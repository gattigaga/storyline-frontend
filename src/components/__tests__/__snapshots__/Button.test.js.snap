// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should renders default 1`] = `
<styled.button
  onClick={[Function]}
>
  Login
</styled.button>
`;
